### PR TITLE
docs(agent): rewrite Agent quickstart

### PR DESCRIPTION
**Pull request by natereid72**
This is a significant PR. It changes the entire doc.

Resolves GitHub issue: <link to GH issue>

Resolves Jira: [JIRAPROJECT-123]

### CODE REVIEW

File: content/en/docs/armory-agent/armory-agent-quick.md
Comment (ncknt) [inline content/en/docs/armory-agent/armory-agent-quick.md]: k8s -> Kubernetes.

File: content/en/docs/armory-agent/armory-agent-quick.md
Comment (ncknt) [inline content/en/docs/armory-agent/armory-agent-quick.md:20]: Why is Redis removed? It's a requirement if you operate more than one clouddriver.

File: content/en/docs/armory-agent/armory-agent-quick.md
Comment (ncknt) [inline content/en/docs/armory-agent/armory-agent-quick.md]: The `spinnaker-kustomize-patches` repo above is a complete set of patches, the expectation is not to use it as is, but to grab example patches as needed. I would not make it a requirement as it's a no go for most users.

I'd say:
- either you use kustomize and just include that patch - it's easy to configure the plugin.
- or you don't use kustomize in which case you'll need to insert the following configuration blocks in `SpinnakerService`.

File: content/en/docs/armory-agent/armory-agent-quick.md
Comment (ncknt) [inline content/en/docs/armory-agent/armory-agent-quick.md]: seems unnecessary since you've configured Clouddriver on the previous step.

File: content/en/docs/armory-agent/armory-agent-quick.md
Comment (ncknt) [inline content/en/docs/armory-agent/armory-agent-quick.md]: I like that we're spelling out gRPC and "from Agent to clouddriver", but I'd leave the http/2 comment. Some people are not familiar with it enough to understand the implication that it won't work with any ingress or proxy they are already using (for the out of cluster case).

File: content/en/docs/armory-agent/armory-agent-quick.md
Comment (ncknt) [inline content/en/docs/armory-agent/armory-agent-quick.md]: k8s -> Kubernetes.

> Running Agent in the target cluster with a service account is the preferred model.

This is not true. There are plenty of reasons to prefer infrastructure or service mode: easier to manage credentials if they are generated, policies forbidding agents, difficulties upgrading Agents as we don't have a good answer for that yet.

File: content/en/docs/armory-agent/armory-agent-quick.md
Comment (ncknt) [inline content/en/docs/armory-agent/armory-agent-quick.md]: `comatible` typo.

File: content/en/docs/armory-agent/armory-agent-quick.md
Comment (ncknt) [inline content/en/docs/armory-agent/armory-agent-quick.md]: The `extensions...` part is actually not needed. We need to remove from the generated manifests as well.

File: content/en/docs/armory-agent/armory-agent-quick.md
Comment (ncknt) [inline content/en/docs/armory-agent/armory-agent-quick.md]: We need to say how to add: as a base? as a resource? as a patch?

I actually preferred when all these details were hidden as they add complexity for no gain.

If you want to show how to configure the plugin without kustomize, then make a separate section. Otherwise I'd keep the kustomization that just shows the thing users can change: version and config.

File: content/en/docs/armory-agent/armory-agent-quick.md
Comment (ncknt) [inline content/en/docs/armory-agent/armory-agent-quick.md]: we usually say `clouddriver` vs `cloud-driver`.

File: content/en/docs/armory-agent/armory-agent-quick.md
Comment (ncknt) [inline content/en/docs/armory-agent/armory-agent-quick.md]: This is not just about remote agents. All agents require a service defined to be able to reach to clouddriver. You can then direct an ingress to the service for remote agents.

Comment (ncknt) [reply content/en/docs/armory-agent/armory-agent-quick.md]: Oh I see the `type: LoadBalancer`. It's missing all the provider annotations (e.g. you'd want an NLB in AWS). Also people sometimes use an http/2 proxy (such as envoy) to perform authz. I'd stay away from attempting to cover all cases and instead explain what it is we need.

File: content/en/docs/armory-agent/armory-agent-quick.md
Comment (natereid72) [reply content/en/docs/armory-agent/armory-agent-quick.md:20]: I think it could be moved to the main doc, to keep the quick start more streamlined and simple.

File: content/en/docs/armory-agent/armory-agent-quick.md
Comment (natereid72) [reply content/en/docs/armory-agent/armory-agent-quick.md]: Ok. Something I just discovered (had flown under my radar all these years) is most of the mainstream ingress controllers can layer 7 grpc. I set this up with L4 LB expose.

File: content/en/docs/armory-agent/armory-agent-quick.md
Comment (natereid72) [reply content/en/docs/armory-agent/armory-agent-quick.md]: I'm going with an assumption that the user knows our kustomize repo, and kustomize. I can reword from adding to kustomize patches, to place in patchesStrategicMerge.

When the details were hidden, it was very confusing. I think we need to find the balance between who we target this for. If someone know K8s, kustomize, and our kustomize repo, they will get this right away. If they don't know those elements, they won't likely have gotten to this point anyway,

File: content/en/docs/armory-agent/armory-agent-quick.md
Comment (natereid72) [reply content/en/docs/armory-agent/armory-agent-quick.md]: ok

File: content/en/docs/armory-agent/armory-agent-quick.md
Comment (natereid72) [reply content/en/docs/armory-agent/armory-agent-quick.md]: when agent is running on same cluster as clouddriver, it can just talk via clusterIP. this is technically a expose method within the cluster. I will think about how to reword this. For simplicity, I'm using the K8s LB config. I can add a note that various cloud providers may require aditional annotation to work.

File: content/en/docs/armory-agent/armory-agent-quick.md
Comment (natereid72) [reply content/en/docs/armory-agent/armory-agent-quick.md]: ok

File: content/en/docs/armory-agent/armory-agent-quick.md
Comment (natereid72) [reply content/en/docs/armory-agent/armory-agent-quick.md]: ok, it's a matter of opinion I think. I can just remove the preferred statement.

File: content/en/docs/armory-agent/armory-agent-quick.md
Comment (natereid72) [reply content/en/docs/armory-agent/armory-agent-quick.md]: I'll reword. Trying to convey that we need a separate target cluster.

File: content/en/docs/armory-agent/armory-agent-quick.md
Comment (natereid72) [reply content/en/docs/armory-agent/armory-agent-quick.md]: We want our users to use this repo and method. trying to get this to a quick start that doesn't go off on multiple angles to explain this way vs that way.

Comment (leefaus) [reply content/en/docs/armory-agent/armory-agent-quick.md]: @ncknt We understand the purpose is to give people options, at the same time we want to promote opinionated configs.  It may mean reworking some of recipes or to move some of the configs to `expert` configuration to keep people from shooting themselves in the foot while they are trying to learn the concepts.

File: content/en/docs/armory-agent/armory-agent-quick.md
Comment (leefaus) [reply content/en/docs/armory-agent/armory-agent-quick.md]: Maybe instead of preferred is to say `the common model we see` and then elaborate that there are additional advanced configurations to enable scalability.

Comment (natereid72) [reply content/en/docs/armory-agent/armory-agent-quick.md]: sounds good. actually, i think more now that this is a main doc topic, then we just ommit the preferred here. so it stays as a very clear quick start

**Comment by aimeeu** [comment]
Deploy preview:  https://deploy-preview-651--armory-docs.netlify.app/docs/armory-agent/armory-agent-quick/

Rather than delete the Kubernetes permission content entirely (since the [PR](https://github.com/armory/docs/pull/629) was merged last week), I created a new Permission page to house that content.

In the Quickstart:

- Overview: added bullet points on what this tutorial covers
- Removed ## Permission section;  bullet point in over view states how permissions are configured; What's Next links to new permissions page
- changed all code snippets to use Prism 
- highlight lines in code snippets
- add subsections instead of adding numbered steps
- add back What's Next section (we need this for multiple reasons)
- add cross links to other docs pages

**Comment by natereid72** [comment]
> Deploy preview: https://deploy-preview-651--armory-docs.netlify.app/docs/armory-agent/armory-agent-quick/
> 
> Rather than delete the Kubernetes permission content entirely (since the [PR](https://github.com/armory/docs/pull/629) was merged last week), I created a new Permission page to house that content.
> 
> In the Quickstart:
> 
> * Overview: added bullet points on what this tutorial covers
> * Removed ## Permission section;  bullet point in over view states how permissions are configured; What's Next links to new permissions page
> * changed all code snippets to use Prism
> * highlight lines in code snippets
> * add subsections instead of adding numbered steps
> * add back What's Next section (we need this for multiple reasons)
> * add cross links to other docs pages

thanks Aimee, looks nice with the prism format. I still believe redis should be removed from the comments. it's an advanced config and not used in the quickstart. It can be explained in the main doc. 

there are a few minor wording changes around loadbalancer that i think would help. I will wait until the current pr is merged before suggesting further changes though.

**Review by aimeeu — APPROVED** [review]

**Pull request merged**
